engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldPreGenerationScreen.java

```diff
@@ -15,31 +15,258 @@
+                        worldGenerator.setWorldSeed(createSeed(selectedWorld));
+                    }
+                    previewGen = new FacetLayerPreview(environment, worldGenerator);
+                    updatePreview();
+                } catch (UnresolvedWorldGeneratorException e) {
+                    e.getMessage();
+                }
+            }
+        });
+
+        WidgetUtil.trySubscribe(this, "reRoll", button -> {
+            worldGenerator.setWorldSeed(createSeed(selectedWorld));
+            getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Ready to roll!", "World seed for " + selectedWorld + " has been changed");
+            updatePreview();
+        });
+
         StartPlayingScreen startPlayingScreen = getManager().createScreen(StartPlayingScreen.ASSET_URI, StartPlayingScreen.class);
-        WidgetUtil.trySubscribe(this, "continue", button ->
-                triggerForwardAnimation(startPlayingScreen)
-        );
+        WidgetUtil.trySubscribe(this, "continue", button -> {
+            startPlayingScreen.setTargetWorld(findWorldByName(), texture, context);
+            triggerForwardAnimation(startPlayingScreen);
+        });
 
         WorldSetupScreen worldSetupScreen = getManager().createScreen(WorldSetupScreen.ASSET_URI, WorldSetupScreen.class);
-
-        WidgetUtil.trySubscribe(this, "config", button ->
-                triggerForwardAnimation(worldSetupScreen)
-        );
+        WidgetUtil.trySubscribe(this, "config", button -> {
+            try {
+                if (!selectedWorld.isEmpty()) {
+                    worldSetupScreen.setWorld(context, findWorldByName());
+                    triggerForwardAnimation(worldSetupScreen);
+                } else {
+                    getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Worlds List Empty!", "No world found to configure.");
+                }
+            } catch (UnresolvedWorldGeneratorException e) {
+                e.getMessage();
+            }
+        });
 
         WidgetUtil.trySubscribe(this, "close", button ->
                 triggerBackAnimation()
         );
     }
+
+    @Override
+    public void onOpened() {
+        try {
+            if (findWorldByName().getWorldGenerator() == null) {
+                worldGenerator = WorldGeneratorManager.createWorldGenerator(findWorldByName().getWorldGeneratorInfo().getUri(), context, environment);
+                findWorldByName().setWorldGenerator(worldGenerator);
+            } else {
+                worldGenerator = findWorldByName().getWorldGenerator();
+            }
+            if (worldGenerator.getWorldSeed().isEmpty()) {
+                worldGenerator.setWorldSeed(createSeed(selectedWorld));
+            }
+            previewGen = new FacetLayerPreview(environment, worldGenerator);
+            updatePreview();
+        } catch (UnresolvedWorldGeneratorException e) {
+            e.getMessage();
```

Comment:
Do a control shift f for all uses of e.getMessage introduced by you, and fix them all.